### DISCUSSION TITLE
Start and Initialization the states containg list of OrderedDict in init method gives error.

### DISCUSSION THREAD

**Discussion by msdian-hub** [pytransitions/transitions#602] [Q&A]
In a class ABCD inside the init method I have variable state consisting of OrderedDict containing list of dictionaries. Basically the state is of module 'transitions' which are imported from class 'machine' and inside state I have orderedDict consisting of list of dictionaries.

class ABCD(Machine)
  def __init__(self,args):
    self.a=opts['a']
    self.a=opts['b']
    self.states=orderedDict([{'name':'start'},{'name':'connection','on_enter':'go'}])

Error:
**Argument of type "list[dict[str, str]]" cannot be assigned to parameter "__iterable" of type "Iterable[list[str]]" in function "__init__"**

 Is the orderedDict not allowed inside init method. How can I deal with that?

Thanks in advance.

**Comment by aleneum** [comment]
`Machine.state` is an internal property used to store `State` objects (in an `OrderedDict`). It must not be reassigned or overridden. I suggest you rename the property/variable for your custom dictionary. If `[{'name':'start'},{'name':'connection','on_enter':'go'}]` are supposed to be configuration data for `States`. You should use the method `Machine.add_states` instead of trying to assign them to `self.states`:
```
class ABCD(Machine):

    def __init__(self, ...):
        # ...
        self.add_states([{'name':'start'},{'name':'connection','on_enter':'go'}])
```
